configuration/configmgr/configmgrlib/SchemaItem.cpp

```diff
@@ -646,3 +647,93 @@ void SchemaItem::setRequiredInstanceComponents(const std::string list)
 {
     m_requiredInstanceComponents = splitString(list, "|");
 }
+
+
+void SchemaItem::validate(Status &status, bool includeChildren, bool includeHiddenNodes) const
+{
+    //
+    // Check the number of environment nodes against the allowed min/max range. Note that since the schema does not
+    // maintain a parent child relationship like the environment, the environment nodes for this schema item need to
+    // be separated by each environment node's parent. Then the number of nodes, by parent, is compared agains the
+    // allowed min/max range
+    std::map<std::string, std::vector<std::shared_ptr<EnvironmentNode>>> envNodesByParent;
```

Comment:
It looks like this is used to track the number of children for a given parent. Could this become ``std::map<std::string,uint32_t>`` to avoid having a std::vector in the map? Granted this is probably less of a concern as vector should have move semantics

Reply:
The observation is a good one. When writing the code, there was the potential need to inspect each of the nodes by type. While that has not materialized, I would prefer to leave the construct the way it is in case that need arises again.